sdks/go/pkg/beam/core/runtime/graphx/serialize.go

```diff
@@ -207,45 +207,45 @@ func encodeFn(u *graph.Fn) (*v1.Fn, error) {
 	case u.Fn != nil:
 		fn, err := encodeUserFn(u.Fn)
 		if err != nil {
-			return nil, fmt.Errorf("bad userfn: %v", err)
+			return nil, fmt.Errorf("failed to encode function %v, bad userfn: %v", u, err)
 		}
 		return &v1.Fn{Fn: fn}, nil
 
 	case u.Recv != nil:
 		t := reflect.TypeOf(u.Recv)
 		k, ok := runtime.TypeKey(reflectx.SkipPtr(t))
 		if !ok {
-			return nil, fmt.Errorf("bad recv: %v", u.Recv)
+			return nil, fmt.Errorf("failed to encode function %v, bad receiver %v", u, u.Recv)
 		}
 		if _, ok := runtime.LookupType(k); !ok {
-			return nil, fmt.Errorf("recv type must be registered: %v", t)
+			return nil, fmt.Errorf("failed to encode function %v, receiver type %v must be registered", u, t)
 		}
 		typ, err := encodeType(t)
 		if err != nil {
-			panic(fmt.Sprintf("bad recv type: %v", u.Recv))
+			panic(fmt.Sprintf("Failed to encode function %v, bad receiver type: %v", u, err))
 		}
 
 		data, err := json.Marshal(u.Recv)
 		if err != nil {
-			return nil, fmt.Errorf("bad userfn: %v", err)
+			return nil, fmt.Errorf("failed to encode function %v, bad userfn: %v", u, err)
```

Comment:
("Failed to encode structural DoFn %v, %T: %v", u, u.Recv, err)

Reply:
Done, with some changes:

("failed to encode structural DoFn %v, failed to marshal receiver %v: %v", u, u.Recv, err)